### PR TITLE
Dphan/qt6 migration

### PR DESCRIPTION
**Pull request by dorotaphanSiili**
PR for checking CI and easier go through changes
should be squashed at the end of review - I kept it with separate commits because of issues I had: with qtpromise and with plugin
for the plugin:
I removed separate apigear feature for that, it is generated with _api repository using add_qml_module. The plugin is now automatically created by qt, and manually adding separate plugin that register types is not recommended. 

Closes # 

## ✅ Checks

- [ ] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

### CODE REVIEW

File: goldenmaster/tb_enum/api/api.cpp
Comment (w4bremer) [inline goldenmaster/tb_enum/api/api.cpp:1]: This is still a really strange diff. Can you explain why it happens with this PR?
I guess some kind of line ending changes, but it really should not be in this PR.

File: goldenmaster/tb_enum/api/api.h [hunk 1]
```diff
@@ -72,11 +74,13 @@ inline QDataStream &operator>>(QDataStream &ds, Enum0::Enum0Enum &obj) {
     return ds;
 }
 
+
```
Comment (w4bremer) [inline goldenmaster/tb_enum/api/api.h:77]: empty new line change should not be here.

File: templates/thirdparty/qtpromise/CMakeLists.txt [hunk 2]
```diff
@@ -2,7 +2,7 @@ include(FetchContent)
 
 FetchContent_Declare(qtpromise
   GIT_REPOSITORY https://github.com/simonbrunel/qtpromise.git
-  GIT_TAG v0.6.0
+  GIT_TAG v0.7.0
```
Comment (w4bremer) [inline templates/thirdparty/qtpromise/CMakeLists.txt:5]: Should we create an issue to move to Qt's own promise at least for tracking why it wasn't done?

File: goldenmaster/examples/olinkclient/CMakeLists.txt [hunk 3]
```diff
@@ -1,7 +1,7 @@
 project(OLinkClient)
 cmake_minimum_required(VERSION 3.20)
 
-find_package(Qt5 REQUIRED COMPONENTS Gui)
+find_package(Qt6 REQUIRED COMPONENTS Gui )
```
Comment (w4bremer) [inline goldenmaster/examples/olinkclient/CMakeLists.txt:4]: whitespace at the end of line should not be added

**Review by w4bremer — CHANGES_REQUESTED** [review]
Only whitespace, new line changes requested, rest looks good!

File: templates/thirdparty/qtpromise/CMakeLists.txt (hunk 2 above)
Comment (dorotaphanSiili) [reply templates/thirdparty/qtpromise/CMakeLists.txt:5]: https://github.com/apigear-io/template-qtcpp/issues/49 created

**Pull request merged**